core/src/main/java/org/apache/iceberg/hadoop/Util.java

```diff
@@ -69,19 +72,28 @@ public static String[] blockLocations(CombinedScanTask task, Configuration conf)
     return locationSets.toArray(new String[0]);
   }
 
-  public static String[] blockLocations(FileIO io, CombinedScanTask task) {
+  public static String[] blockLocations(FileIO io, ScanTaskGroup<?> taskGroup) {
```

Comment:
probably more of a question for my understanding. Iceberg only guarantee compatibility for classes from iceberg-api module, correct?

Reply:
Correct. Only `iceberg-api` has the API / ABI compatibility guarantees.

Reply:
Yep but this is compatible as `CombinedScanTask`  implements `ScanTaskGroup`.